src/rng.cpp

```diff
@@ -1,19 +1,30 @@
-#include "output.h"
 #include "rng.h"
-#include <stdlib.h>
+#include <random>
 
-long rng(long val1, long val2)
+namespace {
+std::mt19937 random_generator {std::random_device{}()};
```

Comment:
Yes, std::mt19937 is Mersinne twister with a specific set of parameters.  But we have no particular requirements for repeatability or cross-platform stability, or anything else really, so "the best default algorithm for the library implementation" fits the bill nicely.